cmd/start.go

```diff
@@ -698,9 +561,228 @@ func (x *Start) Execute(args []string) error {
 	return nil
 }
 
-func getConfigFiles(datafolder string) error {
+func getIPFSRootHash() (*namepb.IpnsEntry, error) {
+	var err error
+	_, ipnskey := namesys.IpnsKeysForID(ipfsNode.Identity)
+	ival, hasherr := ipfsNode.Repo.Datastore().Get(dshelp.NewKeyFromBinary([]byte(ipnskey)))
+	if hasherr != nil {
+		log.Error("get ipns key:", hasherr)
+		return &namepb.IpnsEntry{}, hasherr
+	}
+	val := ival.([]byte)
+	dhtrec := new(recpb.Record)
+	err = proto.Unmarshal(val, dhtrec)
+	if err != nil {
+		log.Error("unmarshal record", err)
+		return &namepb.IpnsEntry{}, err
+	}
+	e := new(namepb.IpnsEntry)
+	err = proto.Unmarshal(dhtrec.GetValue(), e)
+	if err != nil {
+		log.Error("unmarshal record value", err)
+		return &namepb.IpnsEntry{}, err
+	}
+	return e, nil
+}
+
+func getNodeConfig() *ipfscore.BuildCfg {
+	nodeConfig := &ipfscore.BuildCfg{
+		Repo:   nodeRepo,
+		Online: true,
+		ExtraOpts: map[string]bool{
+			"mplex": true,
+		},
+		DNSResolver: dnsResolver,
+		Routing:     DHTOption,
+	}
+
+	if ipfsConfig.Ipns.UsePersistentCache {
+		nodeConfig.ExtraOpts["ipnsps"] = true
+	}
+
+	if onionTransport != nil {
+		nodeConfig.Host = defaultHostOption
+	}
+	return nodeConfig
+}
+
+func defaultHostOption(ctx context.Context, id peer.ID, ps pstore.Peerstore, bwr metrics.Reporter, fs []*net.IPNet, tpt smux.Transport, protec ipnet.Protector, opts *ipfscore.ConstructPeerHostOpts) (p2phost.Host, error) {
```

Comment:
This long method signature seems ready for args to be flattened into a config struct.